src/testModel.ts

```diff
@@ -615,12 +640,22 @@ export class TestModelCollection extends DisposableBase {
   private _didUpdate: vscodeTypes.EventEmitter<void>;
   readonly onUpdated: vscodeTypes.Event<void>;
   private _context: vscodeTypes.ExtensionContext;
+  private _modelsDisposables: vscodeTypes.Disposable[] = [];
```

Comment:
This is still quite confusing, sorry. We should be able to handle this use case without additional complexity. All the objects we operate have nested lifetimes:

- TestModelCollection outlives models that it contains
- TestModels contain respective trace viewers
- There is a way to clear the collection that destroys all the models due to containment
- There is a way to reset the model while maintaining it alive.

I'm not sure why we can't keep things straightforward, but maybe it is the view aspect that makes us keep just one trace viewer open that stands in the way?

Reply:
Current `TestModelCollection` in main branch is not included in the extensions disposables, and I needed to add it to ensure it was properly released in case the extension is deactivated. But `TestModelCollection` dispose in main branch doesn't reset its models, only disposes the `model.onUpdate` event handlers (it uses the `DisposableBase` implementation).

So, to make this step clear, maybe I can do another PR just trying to fix that.

Reply:
Done: https://github.com/microsoft/playwright-vscode/pull/511

Reply:
Sometimes it's better to move one step back: if we keep trace viewer instances in settings we can avoid registering event handlers in `TestModelCollection` and `TestModel`, so basically I'm rolling back changes in those classes.

So:

- TraceViewer implementations become shallow objects once again, with the same lifetime as the extension (right now, it's only `SpawnTraceViewer`, but the same logic will be applied to `EmbeddedTraceViewer` in the next phase)
- They keep a reference to `TestModelCollection` so that they can get the current configuration / server URL
- All event handling that can impact trace viewers (`models.onUpdated` / `showTrace.onChange`, `embeddedTraceViewer.onChange`) are handled in extensions

Let me know if this approach is better

Reply:
It is hard for me to give a good advice without doing the actual work. My naive thought process is as follows:
- User selects test item in the tree
- Extension (controller) handles this change and makes a decision that trace for the test result should be shown
- Extension creates (or reuses) trace viewer for this test via the test model.
- Extension takes owership of the trace viewer and listens for the model to be disabled and/or removed, destroys dangling trace viewer when needed.
- Extension opens trace in the created trace viewer

Reply:
Ok my latest commit no longer makes sense with the changes you introduced. I'll drop it.

Reply:
I don't think I changed anything fundamentally, but you would know better what applies and what does not. Please let me know which patch I should be reviewing!

Reply:
you can review the current one, I already rolled back the "step back" solution.